### ISSUE TITLE
How should I use Obsidian's api?

### ISSUE THREAD

**Issue by Borber** [SilentVoid13/Templater#317]
https://silentvoid13.github.io/Templater/docs/internal-variables-functions/internal-modules/obsidian-module 

There are no more detailed descriptions here. I tried to use tp.obsidian.FileManager.renameFile() and got an error.

Is there a sample script that uses the Obsidian API?

![image](https://user-images.githubusercontent.com/30563826/128445234-221c0dc7-9c93-49d4-9f39-9859c725c975.png)

**Comment by welpdx** [comment]
@Borber there are several ways to rename files.

To use Obsidian API's FileManager class [renameFile() function](https://github.com/obsidianmd/obsidian-api/blob/bde556afa033e909ebfb9fcee8f5ef288276f78f/obsidian.d.ts#L935), try:
```
<%*
let file = await this.app.workspace.getActiveFile(); //get current T file
var filePath = file.path.toString() //get path of Tfile for renameFile()
let newName = await tp.system.prompt("Enter new name: "); //prompt for new file name

if (filePath.includes("/")) {  //if file is nested in some folders
	filePath = filePath.split("/").slice(0, -1).join("/")   //remove last section of path
	filePath = filePath + "/" + newName + ".md"  //add new file name to last section
}
await app.fileManager.renameFile(file,filePath)  //rename
%>
```

To use Obsidian API's Vault class [rename() function](https://github.com/obsidianmd/obsidian-api/blob/bde556afa033e909ebfb9fcee8f5ef288276f78f/obsidian.d.ts#L3038), try:
```
<%*
let file = await this.app.workspace.getActiveFile(); //get current T file
var filePath = file.path.toString() //get path of Tfile for rename()
let newName = await tp.system.prompt("Enter new name: "); //prompt for new file name

if (filePath.includes("/")) {  //if file is nested in some folders
	filePath = filePath.split("/").slice(0, -1).join("/")   //remove last section of path
	filePath = filePath + "/" + newName + ".md"  //add new file name to last section
}
await app.vault.rename(file,filePath)  //rename
%>
```


To use [Templater API's rename()](https://silentvoid13.github.io/Templater/docs/internal-variables-functions/internal-modules/file-module/#:~:text=the%20new%20file%20title), try:
```
<%*
let file = await this.app.workspace.getActiveFile(); //get current T file
let newName = await tp.system.prompt("Enter new name: "); //prompt for new file name
await tp.file.rename(newName)  //rename
%>
```

Because the templater `tp.file.rename()` function doesn't require you to get the path of file, I highly recommend using it. 

#### tl;dr / fyi / to directly answer your question: 
- In your script `tp.obsidian.FileManager.renameFile(file,newPath)` , `FileManager` should be `fileManager`. Idk why the api has misleading capitalizations lol. 
- Also, `tp` is templater's api. To use obsidian's api, use `app.CLASS.FUNCTION`. I'm not a programmer so don't ask me why it works. I just do what works.


Hope that helps.

**Comment by Borber** [comment]
> @Borber there are several ways to rename files.
> 
> To use Obsidian API's FileManager class [renameFile() function](https://github.com/obsidianmd/obsidian-api/blob/bde556afa033e909ebfb9fcee8f5ef288276f78f/obsidian.d.ts#L935), try:
> 
> ```
> <%*
> let file = await this.app.workspace.getActiveFile(); //get current T file
> var filePath = file.path.toString() //get path of Tfile for renameFile()
> let newName = await tp.system.prompt("Enter new name: "); //prompt for new file name
> 
> if (filePath.includes("/")) {  //if file is nested in some folders
> 	filePath = filePath.split("/").slice(0, -1).join("/")   //remove last section of path
> 	filePath = filePath + "/" + newName + ".md"  //add new file name to last section
> }
> await app.fileManager.renameFile(file,filePath)  //rename
> %>
> ```
> 
> To use Obsidian API's Vault class [rename() function](https://github.com/obsidianmd/obsidian-api/blob/bde556afa033e909ebfb9fcee8f5ef288276f78f/obsidian.d.ts#L3038), try:
> 
> ```
> <%*
> let file = await this.app.workspace.getActiveFile(); //get current T file
> var filePath = file.path.toString() //get path of Tfile for rename()
> let newName = await tp.system.prompt("Enter new name: "); //prompt for new file name
> 
> if (filePath.includes("/")) {  //if file is nested in some folders
> 	filePath = filePath.split("/").slice(0, -1).join("/")   //remove last section of path
> 	filePath = filePath + "/" + newName + ".md"  //add new file name to last section
> }
> await app.vault.rename(file,filePath)  //rename
> %>
> ```
> 
> To use [Templater API's rename()](https://silentvoid13.github.io/Templater/docs/internal-variables-functions/internal-modules/file-module/#:~:text=the%20new%20file%20title), try:
> 
> ```
> <%*
> let file = await this.app.workspace.getActiveFile(); //get current T file
> let newName = await tp.system.prompt("Enter new name: "); //prompt for new file name
> await tp.file.rename(newName)  //rename
> %>
> ```
> 
> Because the templater `tp.file.rename()` function doesn't require you to get the path of file, I highly recommend using it.
> 
> #### tl;dr / fyi / to directly answer your question:
> * In your script `tp.obsidian.FileManager.renameFile(file,newPath)` , `FileManager` should be `fileManager`. Idk why the api has misleading capitalizations lol.
> * Also, `tp` is templater's api. To use obsidian's api, use `app.CLASS.FUNCTION`. I'm not a programmer so don't ask me why it works. I just do what works.
> 
> Hope that helps.

Thank you so much, your kindness is beyond words

**Comment by welpdx** [comment]
You're welcome, I'm happy to help. 

If I have solve this issue, please close it. Thanks. 
Also, please use the [Templater Discussions section](https://github.com/SilentVoid13/Templater/discussions) if you would like help using Templater. This section is for issues with the Templater plugin itself. Thanks
